### PR TITLE
refactor: split batchstore Put method to Create and Update

### PR DESCRIPTION
**Pull request by mrekucci**
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/2782)

### CODE REVIEW

File: pkg/postage/batchservice/batchservice.go [hunk 1]
```diff
@@ -113,7 +113,7 @@ func (svc *batchService) Create(id, owner []byte, normalisedBalance *big.Int, de
 		Immutable:   immutable,
 	}
 
-	err := svc.storer.Put(b, normalisedBalance, depth)
+	err := svc.storer.Create(b, normalisedBalance, depth)
```
Comment (istae) [inline pkg/postage/batchservice/batchservice.go:116]: Create does not need the depth arg

**Review by istae — CHANGES_REQUESTED** [review]

File: pkg/postage/batchservice/batchservice.go [hunk 2]
```diff
@@ -141,9 +141,9 @@ func (svc *batchService) TopUp(id []byte, normalisedBalance *big.Int, txHash []b
 		return fmt.Errorf("get: %w", err)
 	}
 
-	err = svc.storer.Put(b, normalisedBalance, b.Depth)
+	err = svc.storer.Update(b, normalisedBalance, b.Depth)
```
Comment (istae) [inline pkg/postage/batchservice/batchservice.go:144]: Consider UpdateValue function

File: pkg/postage/batchservice/batchservice.go [hunk 3]
```diff
@@ -166,7 +166,7 @@ func (svc *batchService) UpdateDepth(id []byte, depth uint8, normalisedBalance *
 	if err != nil {
 		return fmt.Errorf("get: %w", err)
 	}
-	err = svc.storer.Put(b, normalisedBalance, depth)
+	err = svc.storer.Update(b, normalisedBalance, depth)
```
Comment (istae) [inline pkg/postage/batchservice/batchservice.go:169]: Consider UpdateDepth function. These new functions will greatly reduce complexity and any unnecessary decision making in the batchstore package.

Comment (istae) [reply pkg/postage/batchservice/batchservice.go:169]: Also normalisedBalance is not needed as an arg

File: pkg/postage/batchservice/batchservice.go (hunk 1 above)
Comment (istae) [reply pkg/postage/batchservice/batchservice.go:116]: depth arg is not needed

File: pkg/postage/batchservice/batchservice.go (hunk 2 above)
Comment (istae) [reply pkg/postage/batchservice/batchservice.go:144]: b.Depth arg is not needed

**Review by istae — CHANGES_REQUESTED** [review]

**Review by zelig — APPROVED** [review]
> Reviewed 11 of 11 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2782)* status: all files reviewed, 3 unresolved discussions (waiting on @acud, @istae, and @mrekucci)

___
*[pkg/postage/batchservice/batchservice.go, line 144 at r1](https://reviewable.io/reviews/ethersphere/bee/2782#-MuM3YD99itTFV70i0g3-r1-144:-MuUuhS9-7gtxSuyGleZ:b-fqjit0) ([raw file](https://github.com/ethersphere/bee/blob/e4c3e20133af0668f8cd6c541929370eb67ac530/pkg/postage/batchservice/batchservice.go#L144)):*
<details><summary><i>Previously, istae (istae) wrote…</i></summary><blockquote>

b.Depth arg is not needed
</blockquote></details>

it is needed for dilution events or use two distinct functions as Esad suggests

**Review by mrekucci — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2782)* status: 2 of 12 files reviewed, 3 unresolved discussions (waiting on @acud, @istae, @mrekucci, and @zelig)

___
*[pkg/postage/batchservice/batchservice.go, line 116 at r1](https://reviewable.io/reviews/ethersphere/bee/2782#-MuM3YD99itTFV70i0g3-r1-116:-MukSte_5qgt0VNZDaw2:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/e4c3e20133af0668f8cd6c541929370eb67ac530/pkg/postage/batchservice/batchservice.go#L116)):*
<details><summary><i>Previously, istae (istae) wrote…</i></summary><blockquote>

Create does not need the depth arg
</blockquote></details>

Done.

File: pkg/postage/batchservice/batchservice.go (hunk 3 above)
Comment (mrekucci) [reply pkg/postage/batchservice/batchservice.go:169]: Don't want to create too many methods. As a first iteration, this should be enough and we can review it later after the changes to the reserve computation.

**Review by istae — APPROVED** [review]

**Pull request merged**